x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterLookupJoinIT.java

```diff
@@ -0,0 +1,467 @@
+            var remoteCluster2 = executionInfo.getCluster(REMOTE_CLUSTER_2);
+            assertThat(remoteCluster2.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SUCCESSFUL));
+        }
+    }
+
+    public void testLookupJoinMissingLocalIndex() throws IOException {
+        setupClusters(2);
+        populateLookupIndex(REMOTE_CLUSTER_1, "values_lookup", 10);
+
+        var ex = expectThrows(
+            VerificationException.class,
+            () -> runQuery("FROM logs-*,c*:logs-* | EVAL lookup_key = v | LOOKUP JOIN values_lookup ON lookup_key", randomBoolean())
+        );
+        assertThat(ex.getMessage(), containsString("lookup index [values_lookup] is not available in local cluster"));
+
+        // Without local in the query it's ok
+        try (
+            EsqlQueryResponse resp = runQuery(
+                "FROM c*:logs-* | EVAL lookup_key = v | LOOKUP JOIN values_lookup ON lookup_key",
+                randomBoolean()
+            )
+        ) {
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(10));
+            var columns = resp.columns().stream().map(ColumnInfoImpl::name).toList();
+            assertThat(columns, hasItems("lookup_key", "lookup_name", "lookup_tag", "v", "tag", "remote_tag"));
+            assertThat(columns, not(hasItems("local_tag")));
+
+            EsqlExecutionInfo executionInfo = resp.getExecutionInfo();
+            assertThat(executionInfo.getClusters().size(), equalTo(1));
+
+            var remoteCluster = executionInfo.getCluster(REMOTE_CLUSTER_1);
+            assertThat(remoteCluster.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SUCCESSFUL));
+        }
+    }
+
+    public void testLookupJoinMissingKey() throws IOException {
+        setupClusters(2);
+        populateLookupIndex(LOCAL_CLUSTER, "values_lookup", 10);
+        populateLookupIndex(REMOTE_CLUSTER_1, "values_lookup", 10);
+
+        setSkipUnavailable(REMOTE_CLUSTER_1, true);
+        try (
+            // Using local_tag as key which is not present in remote index
+            EsqlQueryResponse resp = runQuery(
+                "FROM logs-*,c*:logs-* | EVAL local_tag = to_string(v) | LOOKUP JOIN values_lookup ON local_tag",
+                randomBoolean()
+            )
+        ) {
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(20));
+            EsqlExecutionInfo executionInfo = resp.getExecutionInfo();
+            assertThat(executionInfo.getClusters().size(), equalTo(2));
+
+            var localCluster = executionInfo.getCluster(LOCAL_CLUSTER);
+            assertThat(localCluster.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SUCCESSFUL));
+            var remoteCluster = executionInfo.getCluster(REMOTE_CLUSTER_1);
+            // FIXME: verify whether we need to skip or succeed here
+            assertThat(remoteCluster.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SUCCESSFUL));
```

Comment:
nit: to be more explicit in the expectations, maybe we should assert that the result will have the rows from the remote index, but the lookup fields will have nulls?